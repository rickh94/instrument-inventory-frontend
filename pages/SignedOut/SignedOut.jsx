import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { API } from 'aws-amplify'
import { RootPaper, SearchResultsList, LoadingHeader } from '../../components'

const SignedOut = ({ showAlert }) => {
  const [isLoading, setLoading] = useState(false)
  const [records, setRecords] = useState([])
  useEffect(() => {
    const getRecords = async () => {
      setLoading(true)
      try {
        const res = await API.get('instrument-inventory', 'filter/signed-out')
        setRecords(res)
      } catch (err) {
        showAlert(err)
      }
      setLoading(false)
    }

    getRecords()
  }, [])
  return (
    <RootPaper>
      <LoadingHeader title="Signed Out Instruments" isLoading={isLoading} />
      <SearchResultsList results={records} />
    </RootPaper>
  )
}

SignedOut.propTypes = {
  showAlert: PropTypes.func.isRequired,
}

export default SignedOut
