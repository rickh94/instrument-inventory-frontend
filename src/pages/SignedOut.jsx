import React, { useState, useEffect, useContext } from 'react'
import { API } from 'aws-amplify'
import { RootPaper, SearchResultsList, LoadingHeader } from '../components'
import { HelpersContext } from '../contexts'

const SignedOut = ({}) => {
  const [isLoading, setLoading] = useState(false)
  const [records, setRecords] = useState([])
  const { showAlert } = useContext(HelpersContext)

  useEffect(() => {
    const getRecords = async () => {
      setLoading(true)
      try {
        const res = await API.get('instrument-inventory', 'filter/signed-out')
        setRecords(res)
      } catch (err) {
        showAlert(err)
      }
      setLoading(false)
    }

    getRecords()
  }, [])
  return (
    <RootPaper>
      <LoadingHeader title="Signed Out Instruments" isLoading={isLoading} />
      <SearchResultsList results={records} />
    </RootPaper>
  )
}

SignedOut.propTypes = {}

export default SignedOut
