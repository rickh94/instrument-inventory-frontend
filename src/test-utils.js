/* eslint-disable react/prop-types */
import React from 'react'
import { MuiThemeProvider, createMuiTheme, indigo } from '@material-ui/core'
import { HelpersContext, SchemaContext } from './contexts'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'

const theme = createMuiTheme({
  palette: {
    primary: indigo,
    secondary: {
      main: '#0A81D1',
    },
  },
})

export const showAlert = jest.fn()
export const setSearchResults = jest.fn()
export const setFilterResults = jest.fn()
export const searchResults = jest.fn()
export const filterResults = jest.fn()

export const TestTheme = ({ children }) => (
  <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
)

export const TestHelpers = ({ children }) => (
  <HelpersContext.Provider
    value={{
      showAlert,
      setSearchResults,
      setFilterResults,
      searchResults,
      filterResults,
    }}
  >
    {children}
  </HelpersContext.Provider>
)

export const TestRouter = ({ children }) => <MemoryRouter>{children}</MemoryRouter>

export const TestSchema = ({ children, schema = { components: { schemas: {} } } }) => (
  <SchemaContext.Provider value={schema}>{children}</SchemaContext.Provider>
)

export const TestEverything = ({ children }) => (
  <TestRouter>
    <TestTheme>
      <TestHelpers>{children}</TestHelpers>
    </TestTheme>
  </TestRouter>
)

// eslint-disable-next-line no-undef
export const flushPromises = () => new Promise(setImmediate)

const customRender = (ui, options, schema = { components: { schemas: {} } }) =>
  render(<TestSchema schema={schema}>{ui}</TestSchema>, {
    wrapper: TestEverything,
    ...options,
  })

export * from '@testing-library/react'

export { customRender as render }
