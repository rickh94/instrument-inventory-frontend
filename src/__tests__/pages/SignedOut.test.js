import React from 'react'
import { cleanup, render, flushPromises } from '../../test-utils'
import SignedOut from '../../pages/SignedOut'
import { API } from 'aws-amplify'

afterEach(cleanup)

describe('<SignedOut />', () => {
  it('matches snapshot', async () => {
    API.get = jest.fn().mockImplementation(() => Promise.resolve([]))
    const { container } = render(<SignedOut />)
    expect(container).toMatchSnapshot()
    await flushPromises()
    expect(API.get).toHaveBeenCalledWith('instrument-inventory', 'filter/signed-out')
  })
})
