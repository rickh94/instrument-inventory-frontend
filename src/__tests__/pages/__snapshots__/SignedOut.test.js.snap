// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignedOut /> matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 WithStyles(ForwardRef(Paper))-root-1 MuiPaper-rounded"
  >
    <div
      class="makeStyles-heading-29"
      data-testid="loading-header"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
      >
        Signed Out Instruments
      </h5>
      <div
        class="MuiCircularProgress-root makeStyles-loader-30 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Instrument
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Size
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Assigned To
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Location
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      />
    </table>
  </div>
</div>
`;
